joplin/base/models/janis_url.py

```diff
@@ -0,0 +1,199 @@
+from django.db import models
+from modelcluster.models import ClusterableModel
+from modelcluster.fields import ParentalKey
+# from wagtail.snippets.models import register_snippet
+
+from .topic_collection_page import TopicCollectionPage
+from .topic_page import TopicPage
+from .department_page import DepartmentPage
+from locations.models import LocationPage
+from .information_page import InformationPage
+from .service_page import ServicePage
+from .guide_page import GuidePage
+from .official_documents_page import OfficialDocumentPage
+from .form_container import FormContainer
+
+
+class FormContainerJanisUrl(ClusterableModel):
+    page = ParentalKey(FormContainer, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class OfficialDocumentPageJanisUrl(ClusterableModel):
+    page = ParentalKey(OfficialDocumentPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class GuidePageJanisUrl(ClusterableModel):
+    page = ParentalKey(GuidePage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class ServicePageJanisUrl(ClusterableModel):
+    page = ParentalKey(ServicePage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class InformationPageJanisUrl(ClusterableModel):
+    page = ParentalKey(InformationPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class LocationPageJanisUrl(ClusterableModel):
+    page = ParentalKey(LocationPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class DepartmentPageJanisUrl(ClusterableModel):
+    page = ParentalKey(DepartmentPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class TopicCollectionPageJanisUrl(ClusterableModel):
+    page = ParentalKey(TopicCollectionPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+class TopicPageJanisUrl(ClusterableModel):
+    page = ParentalKey(TopicPage, related_name='janis_urls')
+    janis_url = models.ForeignKey('base.JanisUrl', verbose_name='URL', related_name='+', on_delete=models.CASCADE)
+
+
+# @register_snippet
```

Comment:
will it later be a snippet?

Reply:
I don't think so.

Reply:
I was thinking about making this a snippet so we could see it without going into a console, same reason I added urls to the author interface, mostly a "hey look at this PR app we can see what urls are being generated" thing

Reply:
We could also just display at as information as part of a panel in that case, or set up a model admin view. My guess  is that it will be weird UX to have the urls listed there and look like content authors are responsible for them when they aren't.

Reply:
agreed that it'd be odd having it where it is now, as I mentioned below I'm not sure if we want to show it at all, I might just remove that part from this PR completely until we have a designed decision.